package/MDAnalysis/core/universe.py

```diff
@@ -494,8 +495,22 @@ def transfer_to_memory(self, start=None, stop=None, step=None,
                 coordinates = []  # TODO: use pre-allocated array
                 for i, ts in enumerate(self.trajectory[start:stop:step]):
                     coordinates.append(np.copy(ts.positions))
+                    dimensions.append(np.copy(ts.dimensions))
                     pm.echo(i, frame=ts.frame)
                 coordinates = np.array(coordinates)
+            # If we extrated the coordinates using Timeseries, we still need
+            # to go through the trajectory to extract the dimensions.
+            else:
+                n_frames = len(range(
+                    *self.trajectory.check_slice_indices(start, stop, step)
+                ))
+                pm_format = ('{step}/{numsteps} frame dimensions '
+                             'copied to memory (frame {frame})')
+                pm = ProgressMeter(n_frames, interval=1,
+                                   verbose=verbose, format=pm_format)
+                for i, ts in enumerate(self.trajectory[start:stop:step]):
```

Comment:
I have no problem using xarray here. And I would be verry happy for the memory reader to expose everything we have.

Though, one issue I see with the feature in general is that `ProtoReader.timeseries` was initially thought as a fallback in case the reader does not already implement a timeseries mechanism (as the DCD and the memory reader do). The fancier we make `ProtoReader.timeseries`, the more difficult it will be to have feature parity among the readers.

In practice: the DCD reader as a `timeseries` method that only returns the positions. I could rename `DCDReader.timeseries` to `_timeseries_positions` and try to use it if available in `ProtoReader.timeseries`, but as soon as the user asks for positions and something else, then I'll have to iterate twice.